core/src/main/java/org/jboss/pnc/rex/core/TaskControllerImpl.java

```diff
@@ -207,8 +207,13 @@ private Transition getTransition(Task task) {
                 if (task.getStopFlag() == StopFlag.CANCELLED)
                     yield Transition.STARTING_to_STOP_REQUESTED;
                 List<ServerResponse> responses = task.getServerResponses().stream().filter(sr -> sr.getState() == State.STARTING).toList();
-                if (responses.stream().anyMatch(ServerResponse::isPositive))
-                    yield Transition.STARTING_to_UP;
+                if (responses.stream().anyMatch(ServerResponse::isPositive)) {
```

Comment:
for cancellation equivalent? I guess also new Transition STOP-REQUESTED -> STOPPED

Reply:
Thanks for the great suggestion!! I completely omitted the stopCallback option. Added in the last commit :)